### ISSUE TITLE
Updating event image should update NFT image if they were the same before update

### ISSUE THREAD

**Issue by snackman** [unlock-protocol/unlock#13545]
**Describe the bug**
When you update an event's image, it doesn't update the NFT image

**To Reproduce**

1. Update event image
2. See that NFT image hasn't changed

**Expected behavior**
If event image and NFT are the same, updating event image should update NFT image. If they're different, it shouldn't (or maybe prompt the user? "Do you want to update the NFT image?"). It's also a little tricky that the NFT image is buried in the lock config section, not obvious for an event manager.

**Screenshots**
![Screenshot_033024_015443_PM](https://github.com/unlock-protocol/unlock/assets/830210/25a825b1-6120-4ffa-8167-b1fe102407b7)

**Desktop (please complete the following information):**

N/A

**Smartphone (please complete the following information):**

N/A

**Additional context**
Apologies if this should be a Feature request and not a Bug report

**Comment by julien51** [comment]
Thanks! This is something we can indeed improve! One of the challenges is that some events have multiple contracts like this one for example https://app.unlock-protocol.com/event/farcon

**Comment by blahkheart** [comment]
@julien51 Is this bounty still open? If so, I can work on it.

**Comment by julien51** [comment]
Yes @blahkheart ! Let's add 250 USDC on that one! It should be pretty straightfroward. Please let me know if you need clarification!

**Comment by 694410194** [comment]
Opened PR #16415 for this bounty: https://github.com/unlock-protocol/unlock/pull/16415

It updates event image saves so lock NFT metadata follows the event image only when the NFT image still matches the previous event image. If a lock already has a custom NFT image, it is preserved.

Verification:
- `NODE_ENV=test ../node_modules/.bin/vitest run __tests__/operations/eventOperations.test.ts --coverage=false` -> 12 tests passed
